psp/models/recent_history.py

```diff
@@ -164,20 +165,23 @@ def __init__(
         """
         super().__init__(config)
         # Validate some options.
-        if nwp_dropout > 0.0 or pv_dropout > 0.0:
+
+        if any(dropout > 0.0 for dropout in nwp_dropout) or pv_dropout > 0.0:
             assert random_state is not None
 
-        if use_nwp:
-            assert nwp_data_source is not None
+        if use_nwp and any(data_source is not None for data_source in use_nwp):
+            assert use_nwp and any(
+                data_source is not None for data_source in use_nwp
+            ), "At least one data_source must be not None"
 
         self._pv_data_source: PvDataSource
-        self._nwp_data_source: NwpDataSource | None
+        self._nwp_data_source: list[NwpDataSource] | None
```

Comment:
```suggestion
        self._nwp_data_source: list[NwpDataSource]
```
Since it's now a list, you can have the empty list as the value for "no nwp", no need for `None` anymore!